### PR TITLE
SecretsManager: Add BatchGetSecretValue implementation

### PR DESCRIPTION
**Pull request by hpaskalev**
Hi all, I've been working on some stuff these days and stumbled upon this [issue](https://github.com/localstack/localstack/issues/9982) on the LocalStack board. 
So, I decided to contribute by providing an implementation for the batch_get_secret_value feature.

### CODE REVIEW

**Comment by hpaskalev** [comment]
The missing tests codecov was complaining about should be now in place.

File: tests/test_secretsmanager/test_secretsmanager.py [hunk 1]
```diff
@@ -181,6 +181,125 @@ def test_get_secret_version_stage_mismatch():
...
+def test_batch_get_secret_value_for_secret_id_list_with_matches():
+    conn = boto3.client("secretsmanager", region_name="us-west-2")
+
+    secret_a = conn.create_secret(Name="test-secret-a", SecretString="secret")
+    secret_b = conn.create_secret(Name="test-secret-b", SecretString="secret")
+
+    secrets_batch = conn.batch_get_secret_value(
+        SecretIdList=["test-secret-a", "test-secret-b"]
+    )
+    matched = [
```
Comment (bblommers) [inline tests/test_secretsmanager/test_secretsmanager.py:194]: This has the potential to hide some problems - `secrets_batch` could (somehow) contain other secrets that are filtered out in this list.

```suggestion
    assert len(secret_batch["SecretValues"]) == 2
```
This achieves the same thing, as far as I can see.

File: tests/test_secretsmanager/test_secretsmanager.py [hunk 2]
```diff
@@ -181,6 +181,125 @@ def test_get_secret_version_stage_mismatch():
...
+    conn = boto3.client("secretsmanager", region_name="us-west-2")
+
+    secret_a = conn.create_secret(Name="test-secret-a", SecretString="secret")
+    secret_b = conn.create_secret(Name="test-secret-b", SecretString="secret")
+    conn.create_secret(Name="test-secret-c", SecretString="secret")
+
+    secrets_batch = conn.batch_get_secret_value(
+        Filters=[{"Key": "name", "Values": [secret_a["Name"], secret_b["Name"]]}]
+    )
+    matched = [
```
Comment (bblommers) [inline tests/test_secretsmanager/test_secretsmanager.py:225]: Same here with the matching.

Alternative:
```
assert [sec["ARN"] for sec in secrets_batch["SecretValues"]] == [secret_a["ARN"], secret_b["ARN"]]
```

File: tests/test_secretsmanager/test_secretsmanager.py [hunk 3]
```diff
@@ -181,6 +181,125 @@ def test_get_secret_version_stage_mismatch():
...
+@mock_aws
+def test_batch_get_secret_value_with_both_secret_id_list_and_filters():
+    conn = boto3.client("secretsmanager", region_name="us-west-2")
+
+    with pytest.raises(ClientError) as exc:
+        conn.batch_get_secret_value(
+            Filters=[{"Key": "name", "Values": ["test-secret-a", "test-secret-b"]}],
+            SecretIdList=["foo", "bar"],
+        )
+        err = exc.value.response["Error"]
```
Comment (bblommers) [inline tests/test_secretsmanager/test_secretsmanager.py:242]: This part is never reached. The `batch_get_secret_value` throws an error, and pytest will ignore any other code within this context manager.

It should be unindented, so it is executed after the context manager finishes:
```suggestion
    err = exc.value.response["Error"]
```

File: tests/test_secretsmanager/test_secretsmanager.py [hunk 4]
```diff
@@ -181,6 +181,125 @@ def test_get_secret_version_stage_mismatch():
...
+
+@mock_aws
+def test_batch_get_secret_value_with_max_results_and_no_filters():
+    conn = boto3.client("secretsmanager", region_name="us-west-2")
+
+    conn = boto3.client("secretsmanager", region_name="us-west-2")
+    with pytest.raises(ClientError) as exc:
+        conn.batch_get_secret_value(MaxResults=10, SecretIdList=["foo", "bar"])
+
+        err = exc.value.response["Error"]
```
Comment (bblommers) [inline tests/test_secretsmanager/test_secretsmanager.py:258]: Same un-indentation here

File: tests/test_secretsmanager/test_secretsmanager.py [hunk 5]
```diff
@@ -181,6 +181,125 @@ def test_get_secret_version_stage_mismatch():
...
+    conn = boto3.client("secretsmanager", region_name="us-east-2")
+
+    secret_a = conn.create_secret(Name="test-secret-a")
+    secret_b = conn.create_secret(Name="test-secret-b")
+
+    with pytest.raises(ClientError) as exc:
+        conn.batch_get_secret_value(
+            Filters=[{"Key": "name", "Values": [secret_a["Name"], secret_b["Name"]]}]
+        )
+        err = exc.value.response["Error"]
```
Comment (bblommers) [inline tests/test_secretsmanager/test_secretsmanager.py:299]: Same un-indentation here

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @hpaskalev, thank you for the PR! Please see my comments

File: moto/secretsmanager/models.py [hunk 6]
```diff
@@ -470,6 +472,47 @@ def get_secret_value(
...
+        if max_results and not filters:
+            raise InvalidParameterException(
+                "'Filters' not specified. 'Filters' must also be specified when 'MaxResults' is provided."
+            )
+
+        if secret_id_list:
+            for secret_id in secret_id_list:
+                # TODO perhaps there should be a check if the secret id is valid identifier
+                # and add an error to the list if not
+                secret = self.secrets.get(secret_id)
```
[the remark this answers is not in the document]
Comment (hpaskalev) [reply moto/secretsmanager/models.py:498]: Sure, I actually used this approach in the beginning but dropped it.

**Review by bblommers — APPROVED** [review]
Great - thank you for adding this to Moto @hpaskalev!

**Pull request merged**